### DISCUSSION TITLE
We shall start require JavaDoc in PRs

### DISCUSSION THREAD

**Discussion by MarcinOrlowski** [logisim-evolution/logisim-evolution#1100] [Dev talk]
I think in process of improving source code quality, it'd be good if we shall start require pull requests to also provide JavaDoc blocks for all the newly introduced methods/attributes and for all the existing methods that were directly modified. I am not talking about complete JavaDoc, with exhaustive behavior documentation, description of all the arguments etc (which that would be good of course). But even adding one line describing purpose of the method would be a step forward. Of course I would exclude all the one-liners or methods that are of very low complexity. Thoughts?

@BFH-ktt1 @davidhutchens @maehne @zdimension @R3dst0ne

**Comment by maehne** [comment]
Personally, I would not enforce this with some automatic check, but make it a recommendation, which we can mention in the PR submission template and ask for in our review comments. I think JavaDoc comments would help most for classes and complex methods with either a lot of code or a lot of arguments. It won't help the code quality if these comments get out of sync with the actual implementation. Therefore, such comments would need to get reviewed along any related code changes. IMHO what is more important is that methods, arguments, and variables have always sensible name. This we can enforce via careful reviews.
